api/v1alpha1/fenceagentsremediationtemplate_webhook.go

```diff
@@ -59,20 +64,55 @@ func (farTemplate *FenceAgentsRemediationTemplate) Default() {
 
 var _ webhook.Validator = &FenceAgentsRemediationTemplate{}
 
+// validateTemplateParameters validates template syntax in shared parameters and collects all errors
+func validateTemplateParameters(spec *FenceAgentsRemediationSpec) error {
+	var validationErrors []error
+
+	// Validate template syntax in shared parameters
+	for paramName, paramValue := range spec.SharedParameters {
+		if _, err := template.ProcessParameterValue(paramValue, "dummy-node-name"); err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("invalid template syntax in shared parameter %s: %w", paramName, err))
+		}
+	}
+
+	return errors.NewAggregate(validationErrors)
+}
+
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (farTemplate *FenceAgentsRemediationTemplate) ValidateCreate() (admission.Warnings, error) {
 	webhookFARTemplateLog.Info("validate create", "name", farTemplate.Name)
-	return validateFAR(&farTemplate.Spec.Template.Spec)
+
+	// Aggregate template validation and FAR validation errors
+	aggregated := errors.NewAggregate([]error{
+		validateTemplateParameters(&farTemplate.Spec.Template.Spec),
+		validateFARTemplateSpec(&farTemplate.Spec.Template.Spec),
+	})
+
+	return admission.Warnings{}, aggregated
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (farTemplate *FenceAgentsRemediationTemplate) ValidateUpdate(old runtime.Object) (admission.Warnings, error) {
 	webhookFARTemplateLog.Info("validate update", "name", farTemplate.Name)
-	return validateFAR(&farTemplate.Spec.Template.Spec)
+
+	// Aggregate template validation and FAR validation errors
+	aggregated := errors.NewAggregate([]error{
+		validateTemplateParameters(&farTemplate.Spec.Template.Spec),
+		validateFARTemplateSpec(&farTemplate.Spec.Template.Spec),
+	})
+
+	return admission.Warnings{}, aggregated
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type
 func (farTemplate *FenceAgentsRemediationTemplate) ValidateDelete() (admission.Warnings, error) {
 	webhookFARTemplateLog.Info("validate delete", "name", farTemplate.Name)
 	return nil, nil
 }
+
+// validateFARTemplateSpec validates the underlying FenceAgentsRemediationSpec
+func validateFARTemplateSpec(spec *FenceAgentsRemediationSpec) error {
```

Comment:
IIUC, `validateFAR` is wrapped inside the new `validateFARTemplateSpec` only to ignore the warning. Couldn't we just do the same inside the old `validateFAR`?

Reply:
I'm not sure I follow.
`validateFAR` is used both to validate the remediation and the template. 
When it validates the remediation the warning isn't ignored and when it validates the template it is.
do you suggest to ignore the warning for both cases ?

Reply:
It isn't clear to me why the warnings are ignored in `validateFARTemplateSpec`. Also, `validateFAR`, does not explicitly ignores warnings, but it just returns an empty object as well, which is kind of the same, IIUC.

If `validateFARTemplateSpec` were to return both warning (as empty object) and error, wouldn't be simpler?

Reply:
Got it, SGTM :+1:

Reply:
LGTM as well :+1: